src/compiler/scala/tools/nsc/settings/MutableSettings.scala

```diff
@@ -70,9 +70,23 @@ class MutableSettings(val errorFn: String => Unit)
       case "" :: xs =>
         loop(xs, residualArgs)
       case x :: xs  =>
+        def error() = {
+          val err         = s"bad option: '$x'"
+          // see if there's an option that looks like it
+          val suffix      = x.stripPrefix("-X").stripPrefix("-Y").stripPrefix("-")
+          val leaders     = List("-", "-X", "-Y")
+          val allprefixes = leaders map (_ + suffix)
```

Comment:
Thx, that was because my first thought was only to show prefixes. After I'd moved on to my second thought, it worked for anything after hyphen, so java yielded -Ygen-javap etc.